### PR TITLE
Execute all rc lines

### LINKED ISSUE #1409: Only some lines of tridactylrc are executed
Ok, so... first of all sorry for opening a new issue (maybe relates to #469), but I think it deserves a separate page. This issue made me crazy.

After three days of frustration I finally shrinked down my `tridactylrc` to this:
```
sanitise tridactyllocal tridactylsync

set storageloc     local " secret feature switcher eastern egg

bind g1 tab 1
bind g3 tab 3 "*
bind g5 tab 5
bind g7 tab 7 "*
bind g2 tab 2
bind g4 tab 4 "*
bind g6 tab 6
bind g8 tab 8 "*
bind g9 tab 9

set searchurls.gh  https://github.com/search?utf8=✓&q= "*

bind J tabprev
bind K tabnext "*
```

I just couldn't imagine before, why my bindings aren't working for days. Guess what happens: only starred lines gets executed (i.e. every second). "Impossible." - I said. If I comment out "storageloc" line, or set it to "sync" it _seems_ to work--however I don't use Firefox sync at all, so I don't know if it should make any difference by the way.

I was curious, so I went one step further to discover what happens to other lines.

```
sanitise tridactyllocal tridactylsync

set storageloc     local " That line from above.

bind g1 tab 1 "+ 1
bind g3 tab 3 "*  6
bind g5 tab 5 "+ 2
bind g7 tab 7 "*  7
bind g2 tab 2 "+ 3
bind g4 tab 4 "*  8
bind g6 tab 6 "+ 4
bind g8 tab 8 "   9
aoesnuhasoneuhbind g9 tab 9

set searchurls.gh  https://github.com/search?utf8=✓&q= "*+

bind J tabprev "    10
bind K tabnext "*+ 5
```

After a few hundreds of restarts here's my result. Sometimes just starred or plus signed, **BUT** sometimes all lines (except "g9" of course) get executed (!!!).(Numbers mean nmap order I saw from `:viewconfig`.)

I don't know if it makes any sense in the context of JavaScript, but as I see it must be some race condition in the background, so I feel that I should also mention that my whole Firefox profile lives in RAM.

### PR DESCRIPTION
**Pull request by bovine3dom**
This hopefully would fix #1409.

However, I can't reproduce #1409 on master with `source testc` - all the lines appear in the config, in both master and this branch.

### CODE REVIEW

File: src/lib/config.ts [hunk 1]
```diff
@@ -1296,9 +1296,10 @@ export async function save(storage: "local" | "sync" = get("storageloc")) {
     // storageobj.set({CONFIGNAME: USERCONFIG})
     const settingsobj = o({})
     settingsobj[CONFIGNAME] = USERCONFIG
-    return storage === "local"
+    storage === "local"
         ? browser.storage.local.set(settingsobj)
         : browser.storage.sync.set(settingsobj)
+    return new Promise(resolve => setTimeout(resolve, 0))
```
Comment (bovine3dom) [inline src/lib/config.ts:1302]: Quoting @farnoy from Matrix:

> I believe you can "fix" config.set by adding await new Promise(resolve => setTimeout(resolve, 0)) after this line https://github.com/tridactyl/tridactyl/pull/2208/files#diff-3d46ae9b8655de6d492285665a944ff8R69
this should effectively wait until onChanged handlers get dispatched before returning control

**Comment by glacambre** [comment]
>However, I can't reproduce #1409 on master with source testc - all the lines appear in the config, in both master and this branch.

Have you tried to downclock your computer?

**Comment by bovine3dom** [comment]
No, that's a good idea. I'll try https://github.com/opsengine/cpulimit 

It looks like the easiest way of simulating a slow disk is https://wiki.archlinux.org/index.php/Trickle and a fake network drive.

**Comment by bovine3dom** [comment]
Definitely doesn't fix #1409 :(

**Pull request closed without merging**